File: src/#1379/solution.go
package solution

type TreeNode struct {
	Val         int
	Left, Right *TreeNode
}

func getTargetCopy(original *TreeNode, cloned *TreeNode, target *TreeNode) *TreeNode {
	if cloned == nil {
		return nil
	}
	if cloned.Val == target.Val {
		return cloned
	}
	Lval := getTargetCopy(original, cloned.Left, target)
	Rval := getTargetCopy(original, cloned.Right, target)
	if Lval != nil {
		return Lval
	} else if Rval != nil {
		return Rval
	} else {
		return nil
	}
}

File: src/#1379/solution.js
function TreeNode(val) {
  this.val = val
  this.left = this.right = null
}

var getTargetCopy = function (original, cloned, target) {
  if (!cloned) {
    return null
  }

  if (cloned.val === target.val) {
    return cloned
  }

  return getTargetCopy(original, cloned.left, target) || getTargetCopy(original, cloned.right, target)
}
